six.py

```diff
@@ -843,6 +843,63 @@ def wrapper(cls):
     return wrapper
 
 
+def ensure_binary(s, encoding='utf-8', errors='strict'):
+    """ A helper function to ensure output is six.binary_type.
+
+    For Python 2:
+      - `unicode` -> encoded to `str`
+      - `str` -> `str`
+
+    For Python 3:
+      - `str` -> encoded to `bytes`
+      - `bytes` -> `bytes`
+    """
+    if not isinstance(s, (text_type, binary_type)):
+        raise TypeError("not expecting type '%s'" % type(s))
+    if isinstance(s, text_type):
+        return s.encode(encoding, errors)
+    else:
```

Comment:
This can be `elif isinstance(s, binary_type):` and then the first `if` statement in this function may be removed.